tests/phpunit/tests/media/wpGetimagesize.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+/**
+ * @group media
+ * @group upload
+ * @requires extension gd
+ * @requires extension exif
+ */
+class Tests_Media_WpGetimagesize extends WP_UnitTestCase {
+
+	public static function wpSetUpBeforeClass( WP_UnitTest_Factory $factory ) {
+		require_once DIR_TESTROOT . '/includes/class-wp-test-stream.php';
+
+		stream_wrapper_register( 'wptestmediawpgetimagesize', 'WP_Test_Stream' );
+		WP_Test_Stream::$data = array(
+			'Tests_Media_WpGetimagesize' => array(
+				'/read.jpg' => file_get_contents( DIR_TESTDATA . '/images/test-image-upside-down.jpg' ),
+				'/exif.jpg' => file_get_contents( DIR_TESTDATA . '/images/2004-07-22-DSC_0008.jpg' ),
+			),
+		);
+	}
+
+	public function test_get_jpeg_size_only() {
+		$size = wp_getimagesize( 'wptestmediawpgetimagesize://Tests_Media_WpGetimagesize/read.jpg' );
+
+		$expected = array(
+			600,
+			450,
+			2,
+			'width="600" height="450"',
+			'bits'     => 8,
+			'channels' => 3,
+			'mime'     => 'image/jpeg',
+		);
+
+		$this->assertSame( $expected, $size );
+	}
+
+	public function test_get_jpeg_with_meta_data_via_stream() {
+		// wp_read_image_metadata gets the meta data via wp_getimagesize().
+		$out = wp_read_image_metadata( 'wptestmediawpgetimagesize://Tests_Media_WpGetimagesize/exif.jpg' );
```

Comment:
@peterwilsoncc The failure here is due to `WP_Test_Stream` not supporting seeking. Discovered by adding this into the test:

```php
var_dump( getimagesize( 'wptestmediawpgetimagesize://Tests_Media_WpGetimagesize/exif.jpg' ) );
```
which returns the following:
```php
getimagesize(): stream does not support seeking
```

Reply:
Thanks so much. I'll put it in without tests and reach out to a few plugin developers to request manual testing.

If the test class can be set up to support seeking then the tests can be added as part of that. 

Thanks again boss!